core/templates/domain/admin/admin-backend-api.service.ts

```diff
@@ -140,8 +140,70 @@ export class AdminBackendApiService {
       });
     });
   }
-}
+  flushCache():Promise<void> {
+    return this.http.post<void>(
+      AdminPageConstants.ADMIN_MEMORY_CACHE_HANDLER_URL, {}
+    ).toPromise();
+  }
+
+  clearSearchIndex():Promise<void> {
+    return this.http.post<void>(
+      AdminPageConstants.ADMIN_HANDLER_URL, {}
+    ).toPromise();
+  }
+
+  regenerateTopicRelatedOpportunities(topicId:string):Promise<void> {
+    return this.http.post<void>(
+      AdminPageConstants.ADMIN_HANDLER_URL, {
+        action: 'regenerate_topic_related_opportunities',
+        topic_id: topicId
+      }
+    ).toPromise();
+  }
+
+  uploadTopicSimilarities(data:string | ArrayBuffer):Promise<void> {
+    return this.http.post<void>(
+      AdminPageConstants.ADMIN_HANDLER_URL, {
+        action: 'upload_topic_similarities',
+        data: data,
+      }
+    ).toPromise();
+  }
 
+  sendDummyMail():Promise<void> {
+    return this.http.post<void>(
+      AdminPageConstants.ADMIN_SEND_DUMMY_MAIL_HANDLER_URL, {}
+    ).toPromise();
+  }
+
+  getMemoryCacheProfile():Promise<void> {
```

Comment:
I was expecting Promise to resolve with some data as it's a getter function can you please check whether the type is correct here?

Reply:
done!